### ISSUE TITLE
ItemsSourceAdapter leaks in Fragments and nested Adapters

### ISSUE THREAD

**Issue by anton-s-morozov** [MugenMvvmToolkit/MugenMvvmToolkit#3]
Our app is Fragment-based. We tried to build a list of polls with a following structure:

```
ListView of CardViews with ( TextView + LinearLayout of ProgressBars )
```

We've quickly found that neither destroying the Fragment, nor explicitly null-ing its DataContext doesn't destroy CardViews and their content, all of which (each individual bound View) remain hanging directly on **Native Stack** GC roots.

Rebuilding nested LinearLayouts while scrolling works fine, only the number of visible CardViews + 1 gets leaked.

By artificially destroying the Activity, I've got it to collect CardViews, TextViews, and LinearLayouts, but views from nested Adapters do not give up their roots.

So that's two issues actually, I guess.

**Comment by vyacheslav-volkov** [comment]
It would be great if you can provide XML-file for a item template.

**Comment by anton-s-morozov** [comment]
Sure!
https://gist.github.com/OrcJMR/ccc3c45bb242f93680da

**Comment by anton-s-morozov** [comment]
By the way, whether it may or may not be helpful, here's a long story of how Stuart Lodge has overcome similar leak in MvvmCross
https://github.com/MvvmCross/MvvmCross/issues/17

**Comment by vyacheslav-volkov** [comment]
I found the cause of the leak, adapters hang in memory as long as the activity is available.
https://github.com/MugenMvvmToolkit/MugenMvvmToolkit/blob/master/Platforms/MugenMvvmToolkit.Android/Binding/Infrastructure/ItemsSourceAdapter.cs#L81
https://github.com/MugenMvvmToolkit/MugenMvvmToolkit/blob/master/Platforms/MugenMvvmToolkit.Android/Binding/Infrastructure/ItemsSourceGeneratorBase.cs#l102
Thank you to report a bug will be fixed in version 4.0.
